dotnet/src/Experimental/Assistants/AssistantBuilder.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI.ChatCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Experimental.Assistants.Extensions;
+using Microsoft.SemanticKernel.Experimental.Assistants.Internal;
+using Microsoft.SemanticKernel.Experimental.Assistants.Models;
+
+namespace Microsoft.SemanticKernel.Experimental.Assistants;
+
+/// <summary>
+/// Fluent builder for initializing an <see cref="IAssistant"/> instance.
+/// </summary>
+public partial class AssistantBuilder
+{
+    private readonly AssistantModel _model;
+    private readonly Dictionary<string, ISKFunction> _functions;
+
+    private string? _apiKey;
+    private Func<HttpClient>? _httpClientProvider;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AssistantBuilder"/> class.
+    /// </summary>
+    public AssistantBuilder()
+    {
+        this._model = new AssistantModel();
+        this._functions = new(12);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IAssistant> BuildAsync(CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrWhiteSpace(this._model.Model))
+        {
+            throw new SKException("Model must be defined for assistant.");
+        }
+
+        if (string.IsNullOrWhiteSpace(this._apiKey))
+        {
+            throw new SKException("ApiKey must be provided for assistant.");
+        }
+
+        return
+            await Assistant.CreateAsync(
+                new OpenAIRestContext(this._apiKey!, this._httpClientProvider),
+                new OpenAIChatCompletion(this._model.Model, this._apiKey!),
+                this._model,
+                this._functions.Values,
+                cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
```

Comment:
inherited from?

Reply:
Oh shoot....comments...got it.  Yeah, there used to be an interface that I wiped (not necessary...restoring now)

Reply:
Brought the comments back